### PR TITLE
Add custom registration form for IG-BCE

### PR DESCRIPTION
**Pull request by rmader**
Not quite done yet

### CODE REVIEW

File: apps/custom_auth/forms.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+            raise forms.ValidationError(
+                "Something is wrong with the setup - please try again later"
+            )
+
+        try:
+            Member.objects.get(member_number=member_number)
+            raise forms.ValidationError(
+                "A member with that number already exists."
+            )
+        except ObjectDoesNotExist:
```
Comment (fuzzylogic2000) [inline apps/custom_auth/forms.py]: I know, I don't strictly need the except, but having just the try makes me cringe.

File: apps/custom_auth/forms.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        birth_date = cleaned_data.get("birth_date")
+
+        self.validateMemberNumberAndDate(member_number, birth_date)
+
+    def save(self, request):
+        user = super(IgbceSignupForm, self).save(request)
+        organisation = Organisation.objects.get(
+            id=settings.SITE_ORGANISATION_ID
+        )
+        additional_info = {
```
Comment (fuzzylogic2000) [inline apps/custom_auth/forms.py]: do I need to tell it, that it has to be json or is putting a dict into a json-field enough?

File: apps/custom_auth/forms.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+            )
+
+    def clean(self):
+        cleaned_data = super().clean()
+        member_number = cleaned_data.get("member_number")
+        birth_date = cleaned_data.get("birth_date")
+
+        self.validateMemberNumberAndDate(member_number, birth_date)
+
+    def save(self, request):
```
Comment (fuzzylogic2000) [inline apps/custom_auth/forms.py]: do we need to make sure that all the saving here is atomic or does the save already handle that?

File: apps/custom_auth/forms.py (hunk 1 above)
Comment (rmader) [reply apps/custom_auth/forms.py]: `Member.objects.filter(member_number=member_number).exists()` should be cleaner

File: apps/custom_auth/forms.py (hunk 3 above)
Comment (rmader) [reply apps/custom_auth/forms.py]: Don't see a race condition here...we can drop the `user.save()` though

File: apps/custom_auth/forms.py (hunk 1 above)
Comment (rmader) [reply apps/custom_auth/forms.py]: And, to add more smartassing: `Member.DoesNotExist` ;)

Comment (fuzzylogic2000) [reply apps/custom_auth/forms.py]: Ah! That's the way!

**Pull request merged**